flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerPutTest.java

```diff
@@ -59,6 +60,110 @@
 
 	private final Random rnd = new Random();
 
+
+	// --- concurrency tests for utility methods which could fail during the put operation ---
+
+	/**
+	 * Checked thread that calls {@link BlobServer#getStorageLocation(BlobKey)}
+	 */
+	public static class ContentAddressableGetStorageLocation extends CheckedThread {
+		private final BlobServer server;
+		private final BlobKey key;
+
+		public ContentAddressableGetStorageLocation(BlobServer server, BlobKey key) {
+			this.server = server;
+			this.key = key;
+		}
+
+		@Override
+		public void go() throws Exception {
+			server.getStorageLocation(key);
```

Comment:
Actually, the job directory is only created once so it doesn't help adding more calls. What could help make this happen more often, is to add more threads to `BlobServerPutTest#testServerContentAddressableGetStorageLocationConcurrent()` otherwise the failure only happens once in a while if the implementation is not thread-safe